### DISCUSSION TITLE
Why you do not use await in queries

### DISCUSSION THREAD

**Discussion by dimitkos** [ChilliCream/hotchocolate#6978] [Q&A] [answered]
Hello i started to read the documentation and I noticed this piece of code in the examples,

public class Query
{
    public Task<Book?> GetBookById(
        [Service] IMongoCollection<Book> collection,
        Guid id)
    {
        return collection.Find(x => x.Id == id).FirstOrDefaultAsync();
    }
}

so why you do not use await collection.Find(x => x.Id == id).FirstOrDefaultAsync(); ??

And general beyond this example I want to call any async service inside the query.

**Comment by PascalSenn** [comment, marked as answer]
The return value from `FirstOrDefaultAsync()` is of type `Task<Book?>`, which means you do not need to await it since it matches the expected return type.

As a result, an async state machine is not compiled around it. 

However, you can still use `async` and `await` if you prefer. 
The difference is negligible, and Hot Chocolate understands both approaches.

**Comment by dimitkos** [comment]
Thank you very much!
